### PR TITLE
Exact path matching ignore path

### LINKED ISSUE #58: Ignore specific files?
## Describe the issue

Hi! I am trying to configure the ShellCheck Action to ignore specific files at specific paths. E.g. if I have a project structure like

```
└── examples
    └── bad.sh
```

I would like to ignore `./examples/bad.sh`. It seems I can use `ignore_paths` to ignore everything under `examples`, or `ignore_names` to ignore any files called `bad.sh`, but it's not clear how to ignore files at specific paths. This is problematic if I have directories with some files to ignore, or file names to ignore, but only in certain directories:

```
├── examples
│   ├── bad.sh
│   └── good.sh
└── good_examples
    └── bad.sh
```

I'm not familiar with GitHub Actions, and though I skimmed the code, I couldn't figure out whether this was supported or not. I tried various configurations, but none seemed to work the way I wanted. This might be a feature request instead of a bug. Any help or advice is appreciated, thanks!

### Links

- **Link to action run:** https://github.com/bi1yeu/shellcheck-gh-action-test/runs/5780506120?check_suite_focus=true
- **Link to action configuration:** https://github.com/bi1yeu/shellcheck-gh-action-test/blob/294aaede0b0367ff63b75de857f456bb706ecc3e/.github/workflows/shellcheck.yml
- tried a whole bunch of configurations on this PR https://github.com/bi1yeu/shellcheck-gh-action-test/pull/1

### PR DESCRIPTION
**Pull request by bi1yeu**
Fixes https://github.com/ludeeus/action-shellcheck/issues/58

Currently, to ignore files from being checked, you can use `ignore_paths` and `ignore_names`, but these inputs can't be used in combination to exclude specific files.

As an example, say you want to exclude the file `./folder_a/maybe_ignore.sh`. Using `ignore_paths: folder_a` would ignore other files in that directory that shouldn't be ignored. Using `ignore_names: maybe_ignore.sh` would ignore files that share the same name as the target file, but which exist in other directories.

```
├── folder_a
│   ├── maybe_ignore.sh
│   └── dont_ignore.sh
└── folder_b
    └── maybe_ignore.sh
```

This change makes the `ignore_paths` input a little more expressive:

- allows specifying a full path to a file like `./folder_a/maybe_ignore.sh`
- allows specifying a glob matcher like `**/maybe_ignore.sh`

I also added more tests for `ignore_paths`. And I removed `testfiles/ignore/ignore.sh`, which looked unused. I can add this back if it was being used as an implicit exclusion test.

### CODE REVIEW

File: .github/workflows/ignore_paths.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
-name: 'ignore_paths'
```
Comment (bi1yeu) [inline .github/workflows/ignore_paths.yml:1]: Sorry if these yaml formatting changes are undesirable -- I used VSCode for editing and it was pretty opinionated.

File: .github/workflows/ignore_paths.yml [hunk 2]
```diff
@@ -15,24 +15,55 @@ jobs:
...
-    - name: Run ShellCheck
-      uses: ./
-      id: check
-      with:
-        ignore_paths: ignore
-
-    - name: Verify check
-      run: | 
-        expect="testfiles/test.bash"
-        notexpect="testfiles/ignore/ignore.bash"
```
Comment (bi1yeu) [inline .github/workflows/ignore_paths.yml:30]: There was no file at `testfiles/ignore/ignore.bash`, so it appears this test was trivially passing.

File: action.yaml [hunk 3]
```diff
@@ -103,6 +103,8 @@ runs:
       id: exclude
```
Comment (bi1yeu) [inline action.yaml:103]: FYI -- I put this bash code into a file and then ran shellcheck on it. There were a number of violations like `SC2296 (error): Parameter expansions can't start with {. Double check syntax.` which I didn't attempt to fix here.

Comment (ludeeus) [reply action.yaml:103]: That's expected, it's not bash, they are for the GitHub action parser.

**Review by ludeeus — APPROVED** [review]
Thanks 👍

**Pull request merged**